From 4e2fff556d3e4b2ff7d5b2a7b3707e693721a317 Mon Sep 17 00:00:00 2001
From: Svitlana Drozd <svitlana_drozd@epam.com>
Date: Tue, 4 Nov 2025 14:34:10 +0200
Subject: [PATCH] dt: Add the range for axi to fix the mipX ranges issue

Xen is unable to retrieve address 1 for the msi-controller, which is
<0xff 0xfffff000 0x00 0x1000>, so device tree generation is failing.
That is why this address was added to the ranges in the node axi.

Signed-off-by: Svitlana Drozd <svitlana_drozd@epam.com>
---
 arch/arm64/boot/dts/broadcom/bcm2712.dtsi | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/broadcom/bcm2712.dtsi b/arch/arm64/boot/dts/broadcom/bcm2712.dtsi
index 58d879004bd0..53c2784f8643 100644
--- a/arch/arm64/boot/dts/broadcom/bcm2712.dtsi
+++ b/arch/arm64/boot/dts/broadcom/bcm2712.dtsi
@@ -432,7 +432,8 @@ axi: axi {
 			 <0x10 0x00000000  0x10 0x00000000  0x01 0x00000000>,
 			 <0x14 0x00000000  0x14 0x00000000  0x04 0x00000000>,
 			 <0x18 0x00000000  0x18 0x00000000  0x04 0x00000000>,
-			 <0x1c 0x00000000  0x1c 0x00000000  0x04 0x00000000>;
+			 <0x1c 0x00000000  0x1c 0x00000000  0x04 0x00000000>,
+			 <0xff 0xfffff000  0xff 0xfffff000  0x00 0x00001000>;
 
 		dma-ranges = <0x00 0x00000000  0x00 0x00000000  0x10 0x00000000>,
 			     <0x10 0x00000000  0x10 0x00000000  0x01 0x00000000>,
-- 
2.43.0

